contracts/peg/BasePSM.sol

```diff
@@ -0,0 +1,339 @@
+        IERC20 _underlyingToken,
+        IPCVDeposit _surplusTarget
+    )
+        OracleRef(
+            params.coreAddress,
+            params.oracleAddress,
+            params.backupOracle,
+            params.decimalsNormalizer,
+            params.doInvert
+        )
+    {
+        underlyingToken = _underlyingToken;
+
+        _setReservesThreshold(_reservesThreshold);
+        _setSurplusTarget(_surplusTarget);
+        _setContractAdminRole(keccak256("PSM_ADMIN_ROLE"));
+    }
+
+    /// @notice withdraw assets from PSM to an external address
+    function withdraw(address to, uint256 amount)
+        external
+        virtual
+        override
+        onlyPCVController
+    {
+        _withdrawERC20(address(underlyingToken), to, amount);
+    }
+
+    /// @notice set the ideal amount of reserves for the contract to hold for redemptions
+    function setReservesThreshold(uint256 newReservesThreshold)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setReservesThreshold(newReservesThreshold);
+    }
+
+    /// @notice set the target for sending surplus reserves
+    function setSurplusTarget(IPCVDeposit newSurplusTarget)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setSurplusTarget(newSurplusTarget);
+    }
+
+    /// @notice helper function to set reserves threshold
+    function _setReservesThreshold(uint256 newReservesThreshold) internal {
+        require(
+            newReservesThreshold > 0,
+            "PegStabilityModule: Invalid new reserves threshold"
+        );
+        uint256 oldReservesThreshold = reservesThreshold;
+        reservesThreshold = newReservesThreshold;
+
+        emit ReservesThresholdUpdate(
+            oldReservesThreshold,
+            newReservesThreshold
+        );
+    }
+
+    /// @notice helper function to set the surplus target
+    function _setSurplusTarget(IPCVDeposit newSurplusTarget) internal {
+        require(
+            address(newSurplusTarget) != address(0),
+            "PegStabilityModule: Invalid new surplus target"
+        );
+        require(
+            newSurplusTarget.balanceReportedIn() == address(underlyingToken),
+            "PegStabilityModule: Underlying token mismatch"
+        );
+
+        IPCVDeposit oldTarget = surplusTarget;
+        surplusTarget = newSurplusTarget;
+
+        emit SurplusTargetUpdate(oldTarget, newSurplusTarget);
+    }
+
+    // ----------- Public State Changing API -----------
+
+    /// @notice send any surplus reserves to the PCV allocation
+    function allocateSurplus() external override {
+        int256 currentSurplus = reservesSurplus();
+        require(
+            currentSurplus > 0,
+            "PegStabilityModule: No surplus to allocate"
+        );
+
+        _allocate(currentSurplus.toUint256());
+    }
+
+    /// @notice function to receive ERC20 tokens from external contracts
+    function deposit() external override {
```

Comment:
This is the `deposit()` method to implement the `IPCVDeposit` interface? I think it's standard to also emit the `Deposit` event

Reply:
This deposit function calls `_allocate`, and in that function `surplusTarget.deposit()` is called, so that would then emit the `Deposit` event, I believe if we emitted it here, that would be emitting 2 `Deposit` events